### ISSUE TITLE
Can fury support serializing enumerations by enumeration name?

### ISSUE THREAD

**Issue by xxd819240366** [apache/fury#1786]
## Is your feature request related to a problem? Please describe.
Can fury support serializing enumerations by enumeration name? Serializing enumeration names like a string serializer. Serializing existing enumerations through ordinal will cause errors if the order is destroyed when adding new enumerations. Maybe can directly reuse the write and read methods in the stringSerializer?
![image](https://github.com/user-attachments/assets/3f43eba3-98c7-4c21-914b-4523ce98ca89)

**Comment by chaokunyang** [comment]
Yes, this make sense, would you like to contribute to it? The implementation is not complex. But we'd like to keep serializing by ordinal as default. And encode enum string by Fury MetaString algorithm using `org.apache.fury.resolver.ClassResolver#getMetaStringResolver`

**Comment by xxd819240366** [comment]
OK, I can try it.
